File: task_equivalents/llm_translations/starcoder/optimized/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd

College = pd.read_csv('data/csv/College.csv', header=True)
College.set_index(College.columns[0], inplace=True)
College['Private01'] = (College['Private'] == 'Yes').astype(int)
College.drop('Private', axis=1, inplace=True)
X = College.loc[:, College.columns != 'Apps'].apply(pd.to_numeric)
Y = College['Apps']
train = College.sample(frac=0.5, random_state=1).index
test = [i for i in College.index if i not in train]
xtrain = X.loc[train]
xtest = X.loc[test]
ytrain = Y.loc[train]
ytest = Y.loc[test]

File: task_equivalents/llm_translations/starcoder/optimized/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(dplyr)
library(readr)
library(caTools)

College <- read_csv('data/csv/College.csv')
school_name <- College$X1
College <- rename(College, school_name = X1)
College <- select(College, -school_name)

College$Private01  <- ifelse(College$Private == 'Yes', 1, 0)
College <- select(College, -Private)

X <- select(College, -Apps) %>% as.data.frame()
Y <- College$Apps

set.seed(1) # Set the seed for reproducibility
split <- sample.split(Y, SplitRatio = 0.5)
xtrain <- subset(X, split == TRUE)
xtest <- subset(X, split == FALSE)
ytrain <- subset(Y, split == TRUE)
ytest <- subset(Y, split == FALSE)
